src/ecl/saveData.ts

```diff
@@ -0,0 +1,247 @@
+import * as path from "path";
+import * as vscode from "vscode";
+import { Workunit, Result } from "@hpcc-js/comms";
+import { eclStatusBar } from "./status";
+import ExcelJS from "exceljs";
+import { createArrayCsvWriter } from "csv-writer";
+
+export enum SaveFileType { UNKNOWN, CSV, WORKBOOK }
+export enum SaveType { UNKNOWN, WORKUNIT, RESULT }
+
+export let saveData: SaveData;
+export class SaveData {
+    private _wu: Workunit;
+    private _saveType: SaveType;
+    private _type: SaveFileType;
+    private _filePath: string;
+    private _resultCount: number;
+    private _numResults: number;
+    private _workbook;
```

Comment:
Missing type

Reply:
That would be ExcelJS.Workbook but that is not an assignable type.